docs/customresourcestate-metrics.md

```diff
@@ -0,0 +1,212 @@
+# Custom Resource State Metrics
+
+This section describes how to add metrics based on the state of a custom resource without writing a custom resource 
+registry and running your own build of KSM.
+
+## Configuration
+
+The `KSM_CUSTOM_RESOURCE_METRICS` environment variable can be used to configure the metrics for custom resources:
```

Comment:
I use the variable as I don't like having to create a ConfigMap, add it as a volume, and mount it just to set a few options, and CLI flags for all the variables would be impossible to read.

I added a flag for the config file location as a middle ground.

I agree something like viper would be an improvement.

Reply:
I also think inlining the config would be a better option here. It is always better to have all the configurations in the same place. If we start spreading it, it will just be harder for users to figure out how their apps are configured.